syweb/webclient/app/components/matrix/matrix-service.js

```diff
@@ -243,118 +273,161 @@ function($http, $window, $timeout, $q) {
             rateLimitMaxMs = ms;
         },
 
-        // Register an user
-        register: function(user_name, password, threepidCreds, useCaptcha) {
+        // Register a user
+        register: function(user_name, password, threepidCreds, captchaResponse, sessionId, bind_email) {
```

Comment:
Please use `camelCase` here: I appreciate that `user_name` isn't, so it isn't clear. I'd prefer to only use `_` when we need to access the event object itself.